packages/now-build-utils/src/types.ts

```diff
@@ -99,3 +99,26 @@ export interface PrepareCacheOptions {
    */
   config: Config;
 }
+
+export interface ShouldServeParams {
+  /**
+   * A path string from a request.
+   */
+  requestPath: string;
+  /**
+   * Name of entrypoint file for this particular build job. Value
+   * `files[entrypoint]` is guaranteed to exist and be a valid File reference.
+   * `entrypoint` is always a discrete file and never a glob, since globs are
+   * expanded into separate builds at deployment time.
+   */
+  entrypoint: string;
+  /**
+   * All source files of the project
+   */
+  files: Files;
+  /**
+   * An arbitrary object passed by the user in the build definition defined
+   * in `now.json`.
+   */
+  config?: Config;
```

Comment:
```suggestion
  config: Config;
```

As discussed in slack, lets make this never undefined

Reply:
IMO this should keep what it is from user's config, it is optional in that case. It would be strange for builder runtime to get a `{}` or something else when the builder config has no `config` attr.

Reply:
Why is that strange? It allows the builder to stop checking for existence which prevents duplication in each builder and should also improve performance a bit.

Reply:
Because we will tell user this is the `config` from builder config in `now.json`.

While we can add that "it's default value is `{}`".

Reply:
We already do this for the `build()` function.

Can you think of a use case where a builder would need to separate `undefined` vs `{}`?

Reply:
Just prefer the way needs less explaining ☺️

Reply:
While I'm not strongly against this either :D